src/ucs/sys/init.c

```diff
@@ -13,8 +14,67 @@
 #include <ucs/async/async.h>
 
 
+/* run-time CPU detection */
+static void UCS_F_NOOPTIMIZE ucs_cpu_detection(void)
+{
+    char str[256];
+    char *p_str;
+    int cpu_flags;
+    struct {
+        const char* flag;
+        ucs_cpu_flag_t value;
+    } *p_flags,
+    cpu_flags_array[] = {
+            { "cmov", UCS_CPU_FLAG_CMOV },
+            { "mmx", UCS_CPU_FLAG_MMX },
+            { "mmx2", UCS_CPU_FLAG_MMX2 },
+            { "sse", UCS_CPU_FLAG_SSE },
+            { "sse2", UCS_CPU_FLAG_SSE2 },
+            { "sse3", UCS_CPU_FLAG_SSE3 },
+            { "ssse3", UCS_CPU_FLAG_SSSE3 },
+            { "sse41", UCS_CPU_FLAG_SSE41 },
+            { "sse42", UCS_CPU_FLAG_SSE42 },
+            { "avx", UCS_CPU_FLAG_AVX },
+            { "avx2", UCS_CPU_FLAG_AVX2 },
+            { NULL, UCS_CPU_FLAG_UNKNOWN },
+    };
+
+    cpu_flags = ucs_arch_get_cpu_flag();
+    if (UCS_CPU_FLAG_UNKNOWN == cpu_flags) {
+        return ;
+    }
+#ifdef CPU_FLAGS
+    strncpy(str, UCS_PP_STR(CPU_FLAGS), sizeof(str) - 1);
+#else
+    strncpy(str, "", sizeof(str) - 1);
+#endif
+
+    p_str = strtok(str, " |\t\n\r");
+    while (p_str) {
+        p_flags = cpu_flags_array;
+        while (p_flags && p_flags->flag) {
+            if (!strcmp(p_str, p_flags->flag)) {
+                if (!(cpu_flags & p_flags->value)) {
+                    fprintf(stderr, "FATAL: ucx library can not be loaded\n"
```

Comment:
please make it print something like this:
"[hostname:pid] FATAL: UCX library cannot be loaded, CPU does not support sse3,avx"